homeassistant/components/tradfri/base_class.py

```diff
@@ -34,12 +40,14 @@ class TradfriBaseClass(Entity):
 
     _attr_should_poll = False
 
-    def __init__(self, device, api, gateway_id):
+    def __init__(
+        self, device: Command, api: Callable[[str], Any], gateway_id: str
+    ) -> None:
         """Initialize a device."""
         self._api = handle_error(api)
-        self._device = None
-        self._device_control = None
-        self._device_data = None
+        self._device: Command | None = None
+        self._device_control: SocketControl = None
+        self._device_data: Socket | None = None
```

Comment:
`_device_data` can be a Socket, a Light or a Blind, right?

Reply:
you are right, but please bear in mind the title of this PR, it is about __init__.py and switch.py, the changes in base_class.py are just the needed parts. 

#56341 addresses base_class.py,

However I have made the change.